src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -80,15 +80,10 @@ public override void OnStartActivity(Activity activity, object payload)
             var request = context.Request;
             if (!this.hostingSupportsW3C || !(this.options.Propagator is TextMapPropagator))
             {
-                var parentContext = new ActivityContext(
-                    activity.Context.TraceId,
-                    activity.ParentSpanId,
-                    activity.Context.TraceFlags,
-                    activity.Context.TraceState,
-                    true);
                 var ctx = this.options.Propagator.Extract(default, request, HttpRequestHeaderValuesGetter);
 
-                if (ctx.ActivityContext.IsValid() && ctx.ActivityContext != parentContext)
+                if (ctx.ActivityContext.IsValid()
+                    && ctx.ActivityContext != new ActivityContext(activity.TraceId, activity.ParentSpanId, activity.ActivityTraceFlags, activity.TraceStateString, true))
```

Comment:
Tweaked this for perf. Before we created parentContext each time. Now it should only do that if we extracted a valid ActivityContext. Also, switched from using `activity.Context` to the props directly because a lot of stack stuff happens going through the property:

https://github.com/dotnet/runtime/blob/5b4facb5743a4d40c0755a6440112b6b1447ec60/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L540